benchmarking/data/simulated_data/mock_student_provider.py

```diff
@@ -16,9 +16,7 @@
 class MockStudentProvider(StudentProvider):
     def __init__(self, settings: MockStudentProviderSettings):
         self.settings = settings
-
-    def get(self) -> List[Student]:
-        return create_mock_students(
+        self.students: List[Student] = create_mock_students(
```

Comment:
This is pretty cool - I think we should leave as is though

It's important that the `.get()` returns a new set of students each time because the students might not be the same and this is on purpose. Various aspects of how we generate students is randomized and to preserve the statistical relevance of benchmarks we want to run multiple trials knowing that the set of students is slightly different.